src/main/java/org/wildfly/naming/client/remote/RemoteNamingService.java

```diff
@@ -47,7 +49,12 @@
  * @author <a href="mailto:fjuma@redhat.com">Farah Juma</a>
  */
 public class RemoteNamingService {
-    private static final int[] SUPPORTED_PROTOCOL_VERSIONS = new int[] { 1, 2 };
+    static final int JAVAEE_PROTOCOL_VERSION = 2;
+    static final int JAKARTAEE_PROTOCOL_VERSION = 3;
+    // Batavia transformer sensible constant - it can start with either "javax." or "jakarta." if transformation was performed
+    private static final String VARIABLE_CONSTANT = "javax.ejb.FAKE_STRING";
+    static final int LATEST_VERSION = VARIABLE_CONSTANT.startsWith("jakarta") ? JAKARTAEE_PROTOCOL_VERSION : JAVAEE_PROTOCOL_VERSION;
```

Comment:
wildfly-naming-client is currently not transformed, since it does not directly reference any EE API.

Reply:
It is being transformed @chengfang .
I can see this constant pool String
  #15 = String             #109          // jakarta.ejb.FAKE_STRING
in transformed file wildfly-naming-client-1.0.16.Final-SNAPSHOT-ee9.jar after WildFly is being built with snapshot of this lib.

Reply:
so we are relying on the consumer of wildfly-naming-client to transform it. WildFly does it, but some other consumers probably don't. I think we should enhance wildfly-naming-client project to produce a jakarta variant itself to be directly used by consumers, now that wildfly-naming-client does reference jakarta values. WDYT?

Reply:
Not yet.

Reply:
I included jakartized version of this library to this PR @chengfang